mne/preprocessing/ica.py

```diff
@@ -830,7 +830,7 @@ def _transform_raw(self, raw, start, stop, reject_by_annotation=False):
         try:
             picks = _picks_to_idx(raw.info, self.ch_names, exclude=[])
             raise_ = False if len(picks) == len(self.ch_names) else True
-        except ValueError:
+        except RuntimeWarning:
```

Comment:
I don't think this is the right fix. The only reason this works (I think) is because we have warnings-as-errors turn on in our tests. For example:
```
>>> import warnings
>>> try:
...     warnings.warn('foo')
... except UserWarning:
...     print('caught')
... 
<stdin>:2: UserWarning: foo
```
i.e., in regular use this pattern will not work. I think the correct/better fix is to leave this, but in the places in the *tests* that were failing, you add an additional context manager like:
```
with pytest.warns(RuntimeWarning, match='not found'):
```

Reply:
Oh right... my bad.. I thought it was an error that was getting raised.. Then yes, in the method `_transform_xxxxx` I will use `_picks_to_idx(raw.info, self.ch_names, exclude=[], allow_empty=True)` which might log the warning if you look for a channel not present in the info, but will return every time an array to compare with `self.ch_names`, the picks used to fit the ICA.

Reply:
Is there a better syntax that I don't know than the nesting of content managers?